lambda_functions/analyzer/analyzer_aws_lib.py

```diff
@@ -23,6 +23,22 @@
 class FileDownloadError(Exception):
     """File can't be downloaded from S3 with a 4XX error code - do not retry."""
 
+def get_file_size_from_s3(bucket_name: str, object_key: str) -> int:
+    """Get the file size of an object in S3.
+
+    Args:
+        bucket_name: S3 bucket name.
+        object_key: S3 object key.
+
+    Returns:
+        The file size in bytes.
+    """
+    if not object_key:
```

Comment:
Isn't this already caught with the changes in main.py? It shouldn't be necessary to check for this again here and in download_from_s3.